testing/fetcher.ts

```diff
@@ -230,6 +231,30 @@ export class AuthenticatingFetcher implements Fetcher {
           headers: {...headers, Authorization: `Basic ${encodedAuth}`},
         };
       }
+      case AuthenticationType.Custom: {
+        const request = {
+          url,
+          body,
+          form,
+          headers,
+        };
+        const {params} = this._credentials as CustomCredentials;
+
+        let requestWithTemplateSubstitutions = JSON.stringify(request);
```

Comment:
Hmm, I might just do replacement on each request component separately. Fields a bit weird/fragile to make a big string out of disparate fields and replace across all of them.

Reply:
hmm ok i was debating between the two approaches. Making it explicit SGTM. Another question I had here was how to handle template substitution inside the URL. Currently how we do it for web basic is we replace the URI encoded version of `{{hello-1234}}`. My inclination was that we should support replacing both the URI encoded flavor as well as the normal `{{}}` variant. LMK if you have other opinions

Reply:
+1 don't feel very comfortable do a replacement on the full string. at least don't want to worry about the corner cases.

Reply:
oh yeah good call on url-encoding the secret value